scipy/linalg/_basic.py

```diff
@@ -146,6 +148,29 @@ def solve(a, b, lower=False, overwrite_a=False,
     array([ True,  True,  True], dtype=bool)
 
     """
+    xp = array_namespace(a, b)
+    if check_finite:
+        a = as_xparray(a, check_finite=True)
+        b = as_xparray(b, check_finite=True)
+    if is_numpy(xp):
+        return _solve(a, b, lower=lower, overwrite_a=overwrite_a,
+                      overwrite_b=overwrite_b, check_finite=False,
+                      assume_a=assume_a, transposed=transposed)
+    if lower:
+        raise ValueError(arg_err_msg("lower"))
+    if assume_a != 'gen':
+        raise ValueError(arg_err_msg("assume_a"))
+    if transposed:
+        raise ValueError(arg_err_msg("transposed"))
+    if hasattr(xp, 'linalg'):
+        return xp.linalg.solve(a, b)
+    a = np.asarray(a)
+    b = np.asarray(b)
+    return xp.asarray(_solve(a, b, check_finite=False))
+
+
+def _solve(a, b, lower=False, overwrite_a=False, overwrite_b=False,
+           check_finite=True, assume_a='gen', transposed=False):
```

Comment:
Ah yes, this can probably be removed.

I don't think that we can get rid of the `_asarray_validated` calls yet but we may be able to if `_lib._array_api.compliance_scipy` is changed a bit. Perhaps we should open an issue to follow on from the discussion in https://github.com/scipy/scipy/pull/19187 and consider this there.